server/reports/commission/commission-report.js

```diff
@@ -26,50 +28,72 @@ export function handleCommissionRptRequest(request, response) {
     response.send("Missing \"startdate\" parameter.");
     return;
   }
-  //TODO: Parse startDate from string
-
+  
   let endDate = request.query.enddate;
   if (!endDate) {
     response.send("Missing \"enddate\" parameter.");
     return;
   }
-  //TODO: Parse endDate from string
+  
 
+  //pass validated input params to report generator
+  let rptGenerator = new CommissionReport();
+  rptGenerator.setStartDate(startDate);
+  //rptGenerator.setEndDate(endDate);
+  //rptGenerator.addFakeSalesDataForTesting();
 
-  // Pass validated input params to report generator
-  let rptJsonRowsAsJson = [
-    {
-      "paramName": "Key:",
-      "value": "Value:",
-    },
-    {
-      "paramName": "format",
-      "value": format,
-    },
-    {
-      "paramName": "startDate",
-      "value": startDate,
-    },
-    {
-      "paramName": "endDate",
-      "value": endDate,
-    },
-  ];
-  //^^^ TODO: Replace this dummy data.
+  rptGenerator.generate().then((results) => {
```

Comment:
I just realized that I should have a Promises .catch statement at the end of this chain. This was my first time using them, so I'm learning